subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/health/gc/GarbageCollectionCheck.java

```diff
@@ -46,20 +49,26 @@ public GarbageCollectionCheck(Clock clock, GarbageCollectorMXBean garbageCollect
 
     @Override
     public void run() {
-        List<MemoryPoolMXBean> memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans();
-        for (MemoryPoolMXBean memoryPoolMXBean : memoryPoolMXBeans) {
-            String poolName = memoryPoolMXBean.getName();
-            if (memoryPoolMXBean.getType() == MemoryType.HEAP && poolName.equals(heapMemoryPool)) {
-                GarbageCollectionEvent latest = heapEvents.latest();
-                long currentCount = garbageCollectorMXBean.getCollectionCount();
-                // There has been a GC event
-                if (latest == null || latest.getCount() != currentCount) {
-                    heapEvents.slideAndInsert(new GarbageCollectionEvent(clock.getCurrentTime(), memoryPoolMXBean.getCollectionUsage(), currentCount));
+        try {
+            List<MemoryPoolMXBean> memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans();
+            for (MemoryPoolMXBean memoryPoolMXBean : memoryPoolMXBeans) {
+                String poolName = memoryPoolMXBean.getName();
+                if (memoryPoolMXBean.getType() == MemoryType.HEAP && poolName.equals(heapMemoryPool)) {
+                    GarbageCollectionEvent latest = heapEvents.latest();
+                    long currentCount = garbageCollectorMXBean.getCollectionCount();
+                    // There has been a GC event
+                    if (latest == null || latest.getCount() != currentCount) {
+                        heapEvents.slideAndInsert(new GarbageCollectionEvent(clock.getCurrentTime(), memoryPoolMXBean.getCollectionUsage(), currentCount));
+                    }
+                }
+                if (memoryPoolMXBean.getType() == MemoryType.NON_HEAP && poolName.equals(nonHeapMemoryPool)) {
+                    nonHeapEvents.slideAndInsert(new GarbageCollectionEvent(clock.getCurrentTime(), memoryPoolMXBean.getUsage(), -1));
                 }
             }
-            if (memoryPoolMXBean.getType() == MemoryType.NON_HEAP && poolName.equals(nonHeapMemoryPool)) {
-                nonHeapEvents.slideAndInsert(new GarbageCollectionEvent(clock.getCurrentTime(), memoryPoolMXBean.getUsage(), -1));
-            }
+        } catch (Throwable t) {
+            // this class is used as task in a scheduled executor service, so it must not throw any throwable,
+            // otherwise the further invocations of this task get automatically and silently cancelled
+            LOGGER.debug("Exception while checking garbage collection", t);
```

Comment:
❓ I think this should be logged as a warning or as an error.

Reply:
I am not sure, these are purely internal errors that a user cannot do anything about. For me they are useful messages only when debugging a specific problem.

Reply:
👍